linera-views/src/scylla_db.rs

```diff
@@ -328,41 +335,121 @@ impl ScyllaDbClientInternal {
         client.1.clone()
     }
 
+    /// Test if a table is present in the database or not
+    pub async fn test_existence_table(
+        session: &Session,
+        table_name: &String,
+    ) -> Result<bool, ScyllaDbContextError> {
+        // We check the way the test can fail. It can fail in different ways.
+        let query = format!("SELECT table_name FROM kv.pairs_{}", table_name);
+
+        // Execute the query
+        let result = session.query(query, &[]).await;
+
+        // The missing table translates into a very specific error that we matched
+        let error_miss1 =
+            QueryError::DbError(Invalid, format!("unconfigured table pairs_{}", table_name));
+        let error_miss2 = QueryError::DbError(
+            Invalid,
+            "Undefined name table_name in selection claus".to_string(),
+        );
+        let error_miss3 = QueryError::DbError(Invalid, "Keyspace kv does not exist".to_string());
+        let Err(error) = result else {
+            // If ok, then the tables
+            return Ok(true);
+        };
+        let error_str = format!("{:?}", error);
+        let error_miss1_str = format!("{:?}", error_miss1);
+        let error_miss2_str = format!("{:?}", error_miss2);
+        let error_miss3_str = format!("{:?}", error_miss3);
+        if error_str == error_miss1_str
+            || error_str == error_miss2_str
+            || error_str == error_miss3_str
```

Comment:
Instead of allocating all these strings, we could first match against `QueryError::DbError(DbError::Invalid, error_message)`, and then compare `error_message` to the `&'static str`s.

Reply:
Yes, that works.